zephyrproject-rtos/zephyr
# boards: arm: Add QEMU Cortex-M4 emulation platform

This commit adds the QEMU Cortex-M4 emulation platform (qemu_cortex_m4)
based on the new `netduinoplus2` machine added in the QEMU 5.0.

The QEMU `netduinoplus2` machine emulates the Netduino Plus 2 hardware
that includes the STM32F405 SoC.

Signed-off-by: Stephanos Ioannidis <root@stephanos.io>

Closes #22870

NOTE:
1. This PR must be merged after the QEMU 5.0 is released, integrated into the Zephyr SDK (zephyrproject-rtos/sdk-ng#192), and the CI is updated to use the new SDK.
2. In order to test this PR, follow the instructions below:
    1. Check out the following branch (which has all the necessary patches pre-applied):
        https://github.com/stephanosio/qemu/commits/zephyr_stm32f405_emu
    2. Build QEMU `arm-softmmu` target (`configure --target-list=arm-softmmu` `make`)
    3. Set`QEMU_BIN_PATH` to the directory that contains the `qemu-system-arm` built above.

TODO:
- [x] Confirm that this PR passes all tests with https://github.com/zephyrproject-rtos/sdk-ng/pull/201
- [x] Enable icount and confirm that this PR passes all tests
